components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthnMissingClaimHandler.java

```diff
@@ -197,6 +209,17 @@ protected PostAuthnHandlerFlowStatus handlePostAuthenticationForMissingClaimsReq
                 for (Map.Entry<String, String> missingClaim : missingClaimMap.entrySet()) {
                     Claim claimObj = claimManager.getClaim(missingClaim.getValue());
                     if (claimObj != null && claimObj.isReadOnly()) {
+                        if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                            Map<String, Object> params = new HashMap<>();
+                            params.put(FrameworkConstants.LogConstants.SERVICE_PROVIDER,
+                                    context.getServiceProviderName());
+                            params.put(FrameworkConstants.LogConstants.TENANT_DOMAIN, context.getTenantDomain());
+                            LoggerUtils.triggerDiagnosticLogEvent(
+                                    FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK, params,
+                                    LogConstants.FAILED, "One or more read-only claim is missing in the requested " +
```

Comment:
can we say which are missing?

Reply:
this loops through the missing claims and throws an error when the first read-only claim is found. We can log that claim here

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/4326/commits/4244d1316c95005cd9d8181ba51c04f371175f2e